kafka-client/src/main/java/dev/responsive/kafka/internal/db/SegmentedWindowFlushManager.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2024 Responsive Computing, Inc.
+ *
+ * This source code is licensed under the Responsive Business Source License Agreement v1.0
+ * available at:
+ *
+ * https://www.responsive.dev/legal/responsive-bsl-10
+ *
+ * This software requires a valid Commercial License Key for production use. Trial and commercial
+ * licenses can be obtained at https://www.responsive.dev
+ */
+
+package dev.responsive.kafka.internal.db;
+
+import dev.responsive.kafka.internal.db.partitioning.Segmenter;
+import dev.responsive.kafka.internal.db.partitioning.Segmenter.SegmentPartition;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+import dev.responsive.kafka.internal.utils.PendingFlushSegmentMetadata;
+import dev.responsive.kafka.internal.utils.WindowedKey;
+
+public abstract class SegmentedWindowFlushManager implements WindowFlushManager<SegmentPartition> {
```

Comment:
Note this is just a rename. The only change is adding the `@Override` on `streamTime()`.